packages/core/src/browser/quick-open/prefix-quick-open-service.ts

```diff
@@ -49,7 +49,7 @@ export interface QuickOpenHandler {
      * the quick open widget matches this handler's prefix.
      * Allows to initialize the model with some initial data.
      */
-    init?(): void;
+    init?(): Promise<void>;
```

Comment:
there is not async implementation, why should it be `Promise<void>`? rather on callee-side `await` should be removed

Reply:
QuickOpenTask.init is async.  I haven't looked in details, but I guess we want to wait for it to complete so that the data model is correctly initialized.

Reply:
`MaybePromise<void>` ?

Reply:
Ah right, I forgot about MaybePromise.  And it's defined as `T | Promise<T>` so it is indeed related to the third bullet in @akosyakov's link.